src/main/java/org/triplea/client/launch/LaunchScreen.java

```diff
@@ -0,0 +1,654 @@
+package org.triplea.client.launch;
```

Comment:
> The launch window is not a 'child' or a detail of swing :)

Agreed that it is not a detail of the "Java Swing UI framework", but is it not a detail of the "TripleA Swing client"?  Maybe it's not clear, but `swing` is effectively the name of the current TripleA client based on Swing in the package hierarchy, while `javafx` is the name for the new TripleA client based on JavaFX.  (As I mentioned in #1945, this is why I don't like technology-specific names for subsystems because it can lead to confusion. :smile:)

My understanding is that we make a decision at startup to launch the Swing client _or_ the JavaFX client based on a feature toggle.  Is your vision that we will initially run some Swing code (e.g. for the main menu) and then may selectively run JFX code for different parts of the client?  That is, there will not be independent Swing and JFX clients, but rather they will be integrated to some degree?

If so, I don't think that's a problem, but we need to be clear that that's the expectation.  I know we ran into some issues in the initial JFX PR because there were both Swing and JFX EDTs running concurrently, which was preventing the application from shutting down (I'm sure there's a way to solve the issue; at the time, we simply chose to ensure no Swing EDT was ever spun up when running the JFX client).

And if that is the route we're going, then the whole idea of having separate root packages for the Swing and JFX clients no longer applies.  Simply because we continue to have a single client that happens to use both Swing and JFX. :smile:

> This may be the time when we start naming the clients we are building, ui I think is the folder level where things went a bit wrong. That probably should be the client "name".

I believe @RoiEXLab's reasoning for adding the intermediate `ui` package was to give some context to devs who may not realize JavaFX is a UI framework (see [here](https://github.com/triplea-game/triplea/pull/1898#discussion_r122577105)).  I don't have a problem with removing it so that the client name is the direct child of the `client` package.

But, depending on the answer to my first question above, these `swing` and `javafx` subpackages may no longer be appropriate, and we simply use `org.triplea.client` as the root for all client code.